app/Http/Controllers/Staff/DonationSubscriptionController.php

```diff
@@ -0,0 +1,160 @@
+            ->where('start_at', '>=', $curDate->toDateString())
+            ->where('is_active', '=', false)
+            ->where('end_at', '>', $curDate->toDateString())
+            ->orderBy('start_at')
+            ->paginate(5);
+        $subscriptionsActive = DonationSubscription::with('user')
+            ->where('is_active', '=', true)
+            ->paginate(25);
+        $subscriptionsInactive = DonationSubscription::with('user')
+            ->where('end_at', '<=', $curDate->toDateString())
+            ->where('is_active', '=', false)
+            ->orderBy('end_at', 'desc')
+            ->paginate(10);
+        $subscriptionsActiveArray = DonationSubscription::join('users', 'users.id', 'donation_subscriptions.user_id')
+            ->where('is_donor', '=', true)
+            ->where('start_at', '<=', $curDate->toDateString())
+            ->where('end_at', '>=', $curDate->toDateString())
+            ->pluck('user_id')
+            ->toArray();
+
+        return view('Staff.donations.subscriptions.index', [
+            'subscriptions_active'     => $subscriptionsActive,
+            'subscriptions_upcoming'   => $subscriptionsUpcoming,
+            'subscriptions_inactive'   => $subscriptionsInactive,
+            'subscriptions_active_arr' => $subscriptionsActiveArray,
+            'curdate'                  => $curDate
+        ]);
+    }
+
+    /**
+     * Edit A VIP Subscription.
+     */
+    public function edit(Request $request, int $id): \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+    {
+        $user = $request->user();
+        abort_unless($user->group->is_admin, 403);
+
+        $curDate = Carbon::now();
+        $vip_sub = DonationSubscription::findOrFail($id);
+
+        return view('Staff.donations.subscriptions.edit', [
+            'vip_sub' => $vip_sub,
+            'curdate' => $curDate,
+            'date'    => $curDate
+        ]);
+    }
+
+    /**
+     * Save a VIP Subscription change.
+     */
+    public function update(Request $request, int $id): \Illuminate\Http\RedirectResponse
+    {
+        $user = $request->user();
+        abort_unless($user->group->is_admin, 403);
+
+        $curDate = Carbon::now();
+        $vip_sub_end_at_old = "";
+        $vip_sub = DonationSubscription::findOrFail($id);
+
+        // Set Dates from Input form
+        $vip_sub->start_at = $request->input('start_at');
+        $vip_sub->end_at = $request->input('end_at');
+
+        $v = validator($vip_sub->toArray(), [
+            'start_at' => 'required|date',
+            'end_at'   => 'required|date|after:start_at',
+        ]);
+
+        if ($v->fails()) {
+            return redirect()->route('staff.donations.subscriptions.index')
+                ->withErrors($v->errors());
+        }
+        $vip_sub->save();
+
+        return redirect()->route('staff.donations.subscriptions.index')
+            ->withSuccess('VIP Subscription Was Updated Successfully!');
+    }
```

Comment:
```suggestion
    public function update(Request $request, DonationSubscription $donationSubscription): \Illuminate\Http\RedirectResponse
    {
        $validated = $request->validate([
            'start_at' => [
                'required',
                'date',
            ],
            'end_at' => [
                'required',
                'date',
                'after:start_at',
            ]
        ])
        
        $donationSubscription->update($validated);

        return redirect()->route('staff.donations.subscriptions.index')
            ->withSuccess('VIP Subscription Was Updated Successfully!');
    }
```

Move this validation array into a form request.